src/index.js

```diff
@@ -83,9 +83,12 @@ export default class Channel {
    */
   dispatch (...args) {
     if (!this._mute) {
-      this._listeners = this._listeners.filter(listener => {
+      const listnersToInvoke = this._listeners.slice();
```

Comment:
As discussed it's better to block adding/removing listeners to channel during dispatch operation.
Please add todo for future.